src/Framework/Analyzer/test/WebApplicationBuilder/DisallowConfigureHostBuilderConfigureWebHostTest.cs

```diff
@@ -0,0 +1,240 @@
+using Microsoft.Extensions.Hosting;
+var builder = WebApplication.CreateBuilder();
+builder.Host./*MM*/ConfigureWebHost(webHostBuilder => { });
+");
+        // Act
+        var diagnostics = await Runner.GetDiagnosticsAsync(source.Source);
+
+        // Assert
+        var diagnostic = Assert.Single(diagnostics);
+        Assert.Same(DiagnosticDescriptors.DoNotUseConfigureWebHostWithConfigureHostBuilder, diagnostic.Descriptor);
+        AnalyzerAssert.DiagnosticLocation(source.DefaultMarkerLocation, diagnostic.Location);
+        Assert.Equal("ConfigureWebHost cannot be used with WebApplicationBuilder.Host", diagnostic.GetMessage(CultureInfo.InvariantCulture));
+    }
+
+    [Fact]
+    public async Task WebApplicationBuilder_HostWithConfigureWebHost_ProducesDiagnostics_OnDifferentLine()
+    {
+        // Arrange
+        var source = TestSource.Read(@"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Hosting;
+var builder = WebApplication.CreateBuilder();
+builder.Host.
+    /*MM*/ConfigureWebHost(webHostBuilder => { });
+");
+        // Act
+        var diagnostics = await Runner.GetDiagnosticsAsync(source.Source);
+
+        // Assert
+        var diagnostic = Assert.Single(diagnostics);
+        Assert.Same(DiagnosticDescriptors.DoNotUseConfigureWebHostWithConfigureHostBuilder, diagnostic.Descriptor);
+        AnalyzerAssert.DiagnosticLocation(source.DefaultMarkerLocation, diagnostic.Location);
+        Assert.Equal("ConfigureWebHost cannot be used with WebApplicationBuilder.Host", diagnostic.GetMessage(CultureInfo.InvariantCulture));
+    }
+
+    [Fact]
+    public async Task WebApplicationBuilder_HostWithConfigureWebHost_ProducesDiagnostics_WhenChained()
+    {
+        // Arrange
+        var source = TestSource.Read(@"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Hosting;
+var builder = WebApplication.CreateBuilder();
+builder.Host
+    ./*MM*/ConfigureWebHost(webHostBuilder => { })
+    .ConfigureServices(services => { });
+");
+        // Act
+        var diagnostics = await Runner.GetDiagnosticsAsync(source.Source);
+
+        // Assert
+        var diagnostic = Assert.Single(diagnostics);
+        Assert.Same(DiagnosticDescriptors.DoNotUseConfigureWebHostWithConfigureHostBuilder, diagnostic.Descriptor);
+        AnalyzerAssert.DiagnosticLocation(source.DefaultMarkerLocation, diagnostic.Location);
+        Assert.Equal("ConfigureWebHost cannot be used with WebApplicationBuilder.Host", diagnostic.GetMessage(CultureInfo.InvariantCulture));
+    }
+
+    [Fact]
+    public async Task WebApplicationBuilder_HostWithConfigureWebHost_DoesNotProduceDiagnostics_WhenChained()
+    {
+        // Arrange
+        var source = TestSource.Read(@"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Hosting;
+var builder = WebApplication.CreateBuilder();
+builder.Host
+    .ConfigureServices(services => { }) // Because ConfigureServices() returns IHostBuilder, the type gets erased
```

Comment:
Happy to remove this test if it doesn't add any value. It started as my check the analyzer _did_ work, but then realising that as the receiver type on `ConfigureWebHost()` became `IHostBuilder` it could no longer be detected.

Alternatively I could try and make it actually fire, but that would mean walking up the invocation chain and trying to find the WebApplicationBuilder.Host at the root, but I'm not sure that's worth adding that additional complexity to try and cover so many possible usage patterns.